make_mesh.py

```diff
@@ -0,0 +1,256 @@
+	img_nifti=nibabel.Nifti1Image(img_data,None,header=header)
+	nibabel.save(img_nifti,file_output)
+	return
+
+#Define the boundin:g box of the data matrix. 
+def get_bounding_slices(img):
+	"""
+	Function to determine the boundaries of the given image.
+	
+	Parameters:
+	-----------
+	img : (X,Y,Z) array
+		Image data matrix of which boundaries are to be determined
+	
+	Returns:
+	--------
+	bbox : (Dim,2) array
+		array  with range of indices through the matrix that contain non-zero entries along each axis
+	
+	"""
+
+	dims = numpy.shape(img)
+	mask = img == 0
+	bbox = []
+	all_axis = numpy.arange(img.ndim)
+	for kdim in all_axis:
+		nk_dim = numpy.delete(all_axis, kdim)
+		mask_i = mask.all(axis=tuple(nk_dim))
+		dmask_i = numpy.diff(mask_i)
+		idx_i = numpy.nonzero(dmask_i)[0]
+		if len(idx_i) != 2:
+			#TODO: see if one boundary has been found, and check that)
+			print("No clear boundary found (no zero entries?) in dimension" + kdim)
+			print("Boundary of data matrix is returned instead")
+			idx_i = [0, dims[kdim]-2]
+		bbox.append([idx_i[0]+1, idx_i[1]+1])
+	return bbox
+
+# Trim image along specified axis, size input = voxel
+def cut_img(img,bbox,size,axis,direction):
+	"""
+	Function to trim an image data matrix.
+
+	Parameters:
+	-----------
+	img: (X,Y,Z) array
+		Image data matrix to be trimmed
+	bbox: (Dim,2) array
+		Integer values for each axis that specify bounding box of image as returend by get_bounding_slices()
+	size: int
+		number of voxels to trim
+	axis: int
+		axis along which to trim (0,1,2)
+	direction: int
+		either trim form beginning af axis inwards or from end of axis inwards
+
+	Returns:
+	---------
+	img :  (X,Y,Z) array
+		trimmed data matrix
+	
+	"""
+
+	dims = numpy.shape(img)
+	ind = bbox[axis-1]
+	if (direction == 0):
+		new_ind = ind[0] + size
+		slc = [slice(None)] * len(img.shape)
+		slc[axis] = slice(0,new_ind)
+	elif (direction == 1):
+		new_ind = ind[1] - size
+		slc = [slice(None)] * len(img.shape)
+		slc[axis] = slice(new_ind,dims[axis])
+	img[tuple(slc)] = 0
+	return img
+
+def f(i, j, k, affine):
+	"""
+	Returns affine transformed coordinates (i,j,k) -> (x,y,z) Use to set correct coordinates and size for the mesh
+	
+	Parameters:
+	-----------
+	i,j,k : int
+		integer coordinates of points in 3D space to be transformed
+	affine: (4,4) array
+		4x4 matrix containing affine transformation information of Nifti-Image
+	
+	Returns:
+	--------
+	x,y,z : int
+		affine transformed coordinates of input points
+	
+	"""
+
+	M = affine[:3, :3]
+	abc = affine[:3, 3]
+	return M.dot([i, j, k]) + abc
+
+#Writes an .obj file for the output of marching cube algorithm. Specify affine if needed in mesh. One = True for faces indexing starting at 1 as opposed to 0. Necessary for Blender/SurfIce
+def write_obj(name,verts,faces,normals,values,affine=None,one=False):
+	"""
+	Write a .obj file for the output of marching cube algorithm
+
+	Parameters:
+	-----------
+	name : string
+		ouput file name
+	verts : (V,3) array
```

Comment:
is this standard notation? what is a `(V,3) array` and how is it different from a `(F,3) array` ? Unless this is some commonly known notation I am not aware of, we should just call these inputs `array` or `array-like`, and document the contents in the natural language description of the parameter (i.e. the lines below).